### PR TITLE
work from 3 PRs that got messy in a single commit

### PR DESCRIPTION
**Pull request by sethzenz**
Includes everything from #557 and #565.  Includes key changes from #564 as a new separate stochastic producer, Systematics/plugins/PhotonSmearStochasticEGMTool.cc – including necessary tweaks for compilation.  Python edits from #564 not yet included, but can be added as we figure out what the master configuration should now be.

@ferriff @musella Your comments appreciated!

### CODE REVIEW

**Comment by sethzenz** [comment]
@ferriff @musella I will accept this PR later this morning.  As far as I can tell it leaves us with Pasquale's configuration, and I have verified that it runs.  This will also let me handle other PRs without worrying about new conflicts.  So we will have a synchronized code base to start from in modifying the configuration appropriately, but we will need to validate either what's in now or what we change it to.  Please let me know in the next few hours if you have comments before I accept.

File: setup.sh [hunk 1]
```diff
@@ -105,6 +105,12 @@ echo "Setting up TnP tools..."
 #git cms-addpkg DataFormats/RecoCandidate
 #git cms-addpkg PhysiscsTools/TagAndProbe
 git cms-merge-topic -u matteosan1:egm_tnp_76X_v1
+git cms-merge-topic -u matteosan1:smearer_76X_test
+git remote add cmssw-ferriff https://github.com/ferriff/cmssw.git
+git fetch cmssw-ferriff
```
Comment (musella) [inline setup.sh]: Could this cherry picks be put into a CMSSW topic branch?

As far as I understand cherry-picks give troubles at remote sites.

**Comment by musella** [comment]
I went through the diffs and all my changes seem to be in place.

File: setup.sh (hunk 1 above)
Comment (sethzenz) [inline setup.sh]: Yes, I'll try to do this later this week.  But I have tested this setup.sh file at IC and it works ok.

Comment (ferriff) [inline setup.sh]: On Tue, Feb 23, 2016 at 01:40:07AM -0800, musella wrote:
> > @@ -105,6 +105,12 @@ echo "Setting up TnP tools..."
> >  #git cms-addpkg DataFormats/RecoCandidate
> >  #git cms-addpkg PhysiscsTools/TagAndProbe
> >  git cms-merge-topic -u matteosan1:egm_tnp_76X_v1
> > +git cms-merge-topic -u matteosan1:smearer_76X_test
> > +git remote add cmssw-ferriff https://github.com/ferriff/cmssw.git
> > +git fetch cmssw-ferriff
> 
> Could this cherry picks be put into a CMSSW topic branch?

They have been included yesterday by @matteosan1 in
matteosan1:smearer_76X_test , so after his confirmation those lines can
be removed by the setup.sh .

**Pull request merged**